python/lsst/ts/hvac/mqtt_client.py

```diff
@@ -24,290 +24,85 @@
-
-    async def vea12P05(self, comando_encendido):
-        """Command VEA 12 on the fifth floor.
-
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
-        """
-        raise salobj.ExpectedError("Not implemented")
-
-    async def vea13P05(self, comando_encendido):
-        """Command VEA 13 on the fifth floor.
-
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
-        """
-        raise salobj.ExpectedError("Not implemented")
-
-    async def vea14P05(self, comando_encendido):
-        """Command VEA 14 on the fifth floor.
-
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
-        """
-        raise salobj.ExpectedError("Not implemented")
-
-    async def vea15P05(self, comando_encendido):
-        """Command VEA 15 on the fifth floor.
-
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
-        """
-        raise salobj.ExpectedError("Not implemented")
+class MqttClient:
+    """Client class to connect to the HVAC MQTT server.
 
-    async def vea16P05(self, comando_encendido):
-        """Command VEA 16 on the fifth floor.
+    Parameters
+    ----------
+    host: `str`
+        The hostname of the MQTT server.
+    port: `int`
+        The port of the MQTT service.
+    """
 
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
-        """
-        raise salobj.ExpectedError("Not implemented")
-
-    async def vea17P05(self, comando_encendido):
-        """Command VEA 17 on the fifth floor.
+    def __init__(self, host, port):
+        self.log = logging.getLogger("MqttClient")
+        self.host = host
+        self.port = port
+        self.client = None
+        self.msgs = asyncio.Queue()
+        self.log.info("MqttClient constructed")
 
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
+    async def connect(self):
+        """Connect the client to the MQTT server.
         """
-        raise salobj.ExpectedError("Not implemented")
-
-    async def vec01P01(self, comando_encendido):
-        """Command VEC 01 on the first floor.
+        self.client = mqtt.Client()
+        self.client.on_message = self.on_message
+        self.client.connect(self.host, self.port)
+        self.client.loop_start()
+        self.client.subscribe(LSST_GENERAL_TOPIC)
+        self.log.info("Connected")
 
-        Parameters
-        ----------
-        comando_encendido: `bool`
-            Switched on or off.
+    async def disconnect(self):
+        """Disconnect the client from the MQTT server.
         """
-        raise salobj.ExpectedError("Not implemented")
+        self.client.loop_stop()
+        self.client.disconnect()
+        self.log.info("Disconnected")
```

Comment:
When the MqttClient connects to the MQTT server, the server dumps all the queued messages or actually the last recorded message for each topic. Therefore clearing the queue on the server side is not needed. I could implement sending out final telemetry but since this only involves the messages up to at most 60 seconds ago so I don't think that matters very much so I will leave this the way it is implemented now.